frontend/src/core/comments/components/AddComment.js

```diff
@@ -25,69 +34,87 @@ export default function AddComments(props: Props) {
         addComment,
     } = props;
 
-    const commentInput: any = React.useRef();
-    const minRows = 1;
-    const maxRows = 6;
+    let [editorState, setEditorState] = React.useState(
+        EditorState.createEmpty()
+    )
+    const editor: any = React.useRef(null);
+    const rawContentState = convertToRaw(editorState.getCurrentContent());
 
-    if (!user) {
-        return null;
+    const onChange = (editorState) => {
+        setEditorState(editorState);
     }
 
-    const handleOnChange = () => {
-        const textAreaLineHeight = 24;
-        commentInput.current.rows = minRows;
-
-        const currentRows = Math.trunc(commentInput.current.scrollHeight / textAreaLineHeight);
-
-        if (currentRows < maxRows) {
-            commentInput.current.rows = currentRows;
-        }
-        else {
-            commentInput.current.rows = maxRows;
-        }
+    function focusEditor() {
+        editor.current.focus();
     }
 
-    const handleOnKeyDown = (event: SyntheticKeyboardEvent<>) => {
-        if (event.keyCode === 13 && event.shiftKey === false) {
-            submitComment(event);
+    function keyBindingFn(e: SyntheticKeyboardEvent<>) {
+        if (e.keyCode === 13 && e.shiftKey === false) {
+          return 'submitOnEnter' 
         }
+        
+        return getDefaultKeyBinding(e)
     }
 
-    const submitComment = (event: SyntheticEvent<>) => {
-        event.preventDefault();
-        const comment = commentInput.current.value;
+    React.useEffect(() => {
+        focusEditor()
+    }, []);
 
-        if (!comment) {
-            return null;
+    const handleKeyCommand = (command: string) => {
+        if (command === 'submitOnEnter') {
+            submitComment();
+            return 'handled';
         }
+        return 'not-handled';
```

Comment:
Interesting - is that a requirement of draftjs to return these values?

Reply:
As I understand it, yes.  Since we are in essence creating a custom key binding to trigger the submit function on `enter`

Docs:  https://draftjs.org/docs/advanced-topics-key-bindings

Reply:
Thanks for the pointer, it makes sense now!